lib/iris/fileformats/grib/_load_convert.py

```diff
@@ -2002,6 +2007,35 @@ def product_definition_template_31(section, metadata, rt_coord):
         metadata['aux_coords_and_dims'].append((rt_coord, None))
 
 
+def product_definition_template_40(section, metadata, frt_coord):
+    """
+    Translate template representing an analysis or forecast at a horizontal
+    level or in a horizontal layer at a point in time for atmospheric chemical
+    constituents.
+
+    Updates the metadata in-place with the translations.
+
+    Args:
+
+    * section:
+        Dictionary of coded key/value pairs from section 4 of the message.
+
+    * metadata:
+        :class:`collectins.OrderedDict` of metadata.
+
+    * frt_coord:
+        The scalar forecast reference time :class:`iris.coords.DimCoord`.
+
+    """
+    # Perform identical message processing.
+    product_definition_template_0(section, metadata, frt_coord)
+
+    constituent_type = section['constituentType']
+
+    # Add the constituent type as  an attribute.
+    metadata['attributes']['constituent_type'] = constituent_type
```

Comment:
may I suggest we make this name more specific, perhaps 
`WMO_constituent_type`
or
`GRIB_constituent_type`
?

especially as a code table value is being stored